### PR TITLE
Catch unhandled errors and fix warnings

### PR DESCRIPTION
**Pull request by sverdlov93**
- [ ] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----

1. Add an error assertion for client test deferred function calls.
2. Catch unhandled errors.
3. Fix some warnings for variables that collide with imported package names.

### CODE REVIEW

File: tests/artifactorydownload_test.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func flatDownload(t *testing.T) {
 	if err != nil {
 		t.Error(err)
 	}
-	defer os.RemoveAll(workingDir)
+	defer assert.NoError(t, os.RemoveAll(workingDir))
```
Comment (Or-Geva) [inline tests/artifactorydownload_test.go:37]: `RemoveAll` gets executed, we need to wrap it with func

Comment (sverdlov93) [reply tests/artifactorydownload_test.go:37]: done

File: artifactory/services/movecopy.go [hunk 2]
```diff
@@ -50,7 +50,10 @@ func (mc *MoveCopyService) MoveCopyServiceMoveFilesWrapper(moveSpecs ...MoveCopy
 	moveReaders := []*ReaderSpecTuple{}
 	defer func() {
 		for _, readerSpec := range moveReaders {
-			readerSpec.Reader.Close()
+			e := readerSpec.Reader.Close()
+			if err == nil && e != nil {
```
Comment (eyalbe4) [inline artifactory/services/movecopy.go:54]: Can be simplified to - 
```
if err == nil {
```

File: utils/io/fileutils/temp_test.go [hunk 3]
```diff
@@ -57,3 +53,8 @@ func TestExtractTimestamp(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, int64(8652489), timeStamp.Unix())
 }
+
+func CheckIfFileExistsAndAssert(t *testing.T, name string) {
```
Comment (eyalbe4) [inline utils/io/fileutils/temp_test.go:57]: CheckIfFileExistsAndAssert --> AssertFileExists

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: artifactory/services/movecopy.go (hunk 2 above)
Comment (sverdlov93) [reply artifactory/services/movecopy.go]: its a code inside a deferred loop, so no checking e would override e

File: utils/io/fileutils/temp.go [hunk 4]
```diff
@@ -97,13 +99,25 @@ func CleanOldDirs() error {
...
+	timeStampInt, err := strconv.ParseInt(timestampStr, 10, 64)
 	if err != nil {
 		return time.Time{}, errorutils.CheckError(err)
 	}
 	// Convert to time type.
-	return time.Unix(timeStampint, 0), nil
+	return time.Unix(timeStampInt, 0), nil
+}
+
+func CreateTempDirWithCallbackAndAssert(t *testing.T) (string, func()) {
```
Comment (Or-Geva) [inline utils/io/fileutils/temp.go:113]: I think testing.T struct should be present only in test files.

File: utils/io/fileutils/temp.go [hunk 5]
```diff
@@ -97,13 +99,25 @@ func CleanOldDirs() error {
...
+	tempDirPath, err := CreateTempDir()
+	assert.NoError(t, err, "Couldn't create temp dir")
+	return tempDirPath, func() {
+		RemoveTempDirAndAssert(t, tempDirPath)
+	}
+}
+
+func RemoveTempDirAndAssert(t *testing.T, dirPath string) {
+	assert.NoError(t, RemoveTempDir(dirPath), "Couldn't create temp dir")
```
Comment (Or-Geva) [inline utils/io/fileutils/temp.go:122]: Couldn't create... -> Couldn't remove

File: utils/tests/utils.go [hunk 6]
```diff
@@ -124,5 +117,48 @@ func InitVcsSubmoduleTestDir(t *testing.T, srcPath string) (submodulePath, tmpDi
...
+	pwd, err := os.Getwd()
+	require.NoError(t, err)
+	ChangeDirAndAssert(t, dirPath)
+	return func() {
+		ChangeDirAndAssert(t, pwd)
+	}
+}
+
+func GetwdAndAssert(t *testing.T) string {
+	wd, err := os.Getwd()
```
Comment (Or-Geva) [inline utils/tests/utils.go:138]: This will return the same 'wd' for every call. Let's fix this.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

**Pull request merged**